pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -58,7 +58,7 @@ type volumeService interface {
 	createVolume(opts volumeCreateOpts) (string, string, error)
 	getVolume(volumeID string) (Volume, error)
 	deleteVolume(volumeName string) error
-	expandVolume(volumeID string, newSize int) error
+	expandVolume(volumeID string, newSize int, volumeStatus string) error
```

Comment:
Everything in cloudprovider code related to volumes is dead code and it is here just because it was 1:1 copied from in-tree plugin that handled volumes. It might be good to delete whole openstack_volumes.go from cloudprovider as it is just confusing.

Reply:
@mape90 could you please elaborate? This code is used in openstack-cloud-controller-manager.

Reply:
But openstack-cloud-controller-manager never manages volumes. So any code related to volumes is dead code. The CSI is managing all volume related actions.
In-tree version of openstack-cloud-controller-manager volumes where managed by it and thats why there is the code. However when this repo was done the code was just copy pasted from in-tree to here without any modifications. Then of course the volume control loops where removed. However the code base was never cleaned. There are also a lot of more dead code in openstack-cloud-controller-manager as it now behaves differently than it did when it was in-tree.

Reply:
Now I see, I had to recompile kubelet with the updated cloud-provider dependency to test the live volume resize with the 3.42 microversion. That is why I though that the code is used. When the `k8s.io/pkg/volume/cinder` is going to be removed then?

Reply:
Let's keep this as is and remove the code as a separate PR.

Reply:
@kayrus agreed propose a separate  PR for removal, and also do not make any changes in the pkg `pkg/cloudprovider/providers/openstack/openstack_volumes.go`